### PR TITLE
xrootd :: treat the case of wrong python dir location, and guards imp…

### PR DESCRIPTION
**Pull request by adriansev**
…rovement

### CODE REVIEW

**Comment by adriansev** [comment]
all failed test exhibit:
```
--
29/55 Test #31: testUserCodeInterface ...................***Failed    0.94 sec
Running 1 test case...
[INFO] Init CcdApi with UserAgentID: alientest02-1660020541-7uQVKb, Host: ccdb-test.cern.ch:8080
[ERROR] Requested resource does not exist: ccdb-test.cern.ch:8080/qc/TST/MO/QcTask/example/1660020541202/
```

would be best if this can be fixed to restart the tests and have an all green status

File: xrootd.sh [hunk 1]
```diff
@@ -72,32 +72,51 @@ cmake "$BUILDDIR"                                                     \
...
+        find $INSTALLROOT/lib/python/ -name "*.so" -exec install_name_tool -add_rpath ${INSTALLROOT}/lib {} \;
+        find $INSTALLROOT/lib/ -name "*.dylib" -exec install_name_tool -add_rpath ${INSTALLROOT}/lib {} \;
+      ;;
   esac
-fi
 
+    # Print found XRootD python bindings
+    # just run the the command as this is under "bash -e"
+    echo -ne ">>>>>>   Found XRootD python bindings: "
+    LD_LIBRARY_PATH="$INSTALLROOT/lib${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH" PYTHONPATH="$INSTALLROOT/lib/python/site-packages${PYTHONPATH:+:}$PYTHONPATH" ${PYTHON_EXECUTABLE} -c 'from XRootD import client as xrd_client;print(f"{xrd_client.__version__}\n{xrd_client.__file__}");'
```
Comment (ktf) [inline xrootd.sh:116]: In general, please avoid such long lines.

**Comment by ktf** [comment]
Thanks for your contribution!
For next time, please try to keep the line length down.

File: xrootd.sh (hunk 1 above)
Comment (adriansev) [reply xrootd.sh:116]: Sure, sorry about that! Would you prefer to fix it now, or given the time for the tests to run just let the tests finish, merge if green, and make the line splitting in a new PR? (maybe in the next one with libxml2?)

Comment (ktf) [reply xrootd.sh:116]: Yes, that's fine. No need to make a PR just for that.

**Comment by adriansev** [comment]
Hi @ktf ! i think that those 3 tests are stuck as they are in "expected" for some time..

**Comment by adriansev** [comment]
Hi @ktf ! do you have any idea if the CTest error have anything to do with python bindings of xrootd? it would seem that the error is:
```
[25871:internal-dpl-injected-dummy-sink]: o2-test-framework-SimpleDataProcessingDevice01: /sw/SOURCES/O2/build_O2_alidist-cs8-slc8_x86-64/0/Framework/Core/src/DataProcessingDevice.cxx:1478: o2::framework::DataProcessingDevice::handleData(o2::framework::DataProcessorContext&, o2::framework::InputChannelInfo&)::<lambda()>: Assertion `std::accumulate(results.begin(), results.end(), 0, [](size_t const& count, auto const& element) -> size_t { return count + element.size; })' failed.
[25871:internal-dpl-injected-dummy-sink]: *** Program crashed (Segmentation fault, FPE, BUS, ABRT, KILL)
[25871:internal-dpl-injected-dummy-sink]: Backtrace by DPL:
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: ()
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: gsignal()
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: abort()
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: ()
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: ()
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: ()
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: o2::framework::DataProcessingDevice::handleData(o2::framework::DataProcessorContext&, o2::framework::InputChannelInfo&)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: o2::framework::DataProcessingDevice::doPrepare(o2::framework::DataProcessorContext&)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: o2::framework::run_callback(uv_work_s*)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: o2::framework::DataProcessingDevice::Run()
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQ.so.1.4: fair::mq::Device::RunWrapper()
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQStateMachine.so.1.4: boost::detail::function::void_function_obj_invoker1<std::function<void (fair::mq::State)>, void, fair::mq::State>::invoke(boost::detail::function::function_buffer&, fair::mq::State)
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQStateMachine.so.1.4: boost::signals2::detail::signal_impl<void (fair::mq::State), boost::signals2::optional_last_value<void>, int, std::less<int>, boost::function<void (fair::mq::State)>, boost::function<void (boost::signals2::connection const&, fair::mq::State)>, boost::signals2::mutex>::operator()(fair::mq::State)
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQStateMachine.so.1.4: fair::mq::fsm::Machine_::ProcessWork()
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQStateMachine.so.1.4: fair::mq::StateMachine::ProcessWork()
[25871:internal-dpl-injected-dummy-sink]:     /sw/slc8_x86-64/FairMQ/v1.4.52-6/lib/libFairMQ.so.1.4: fair::mq::DeviceRunner::Run()
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: doChild(int, char**, o2::framework::ServiceRegistry&, o2::framework::RunningWorkflowInfo const&, o2::framework::RunningDeviceRef, o2::framework::ProcessingPolicies, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, uv_loop_s*)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: runStateMachine(std::vector<o2::framework::DataProcessorSpec, std::allocator<o2::framework::DataProcessorSpec> > const&, WorkflowInfo const&, std::vector<o2::framework::DataProcessorInfo, std::allocator<o2::framework::DataProcessorInfo> > const&, o2::framework::CommandInfo const&, o2::framework::DriverControl&, o2::framework::DriverInfo&, std::vector<o2::framework::DeviceMetricsInfo, std::allocator<o2::framework::DeviceMetricsInfo> >&, boost::program_options::variables_map&, std::vector<o2::framework::ServiceSpec, std::allocator<o2::framework::ServiceSpec> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/lib64/libO2Framework.so: doMain(int, char**, std::vector<o2::framework::DataProcessorSpec, std::allocator<o2::framework::DataProcessorSpec> > const&, std::vector<o2::framework::ChannelConfigurationPolicy, std::allocator<o2::framework::ChannelConfigurationPolicy> > const&, std::vector<o2::framework::CompletionPolicy, std::allocator<o2::framework::CompletionPolicy> > const&, std::vector<o2::framework::DispatchPolicy, std::allocator<o2::framework::DispatchPolicy> > const&, std::vector<o2::framework::ResourcePolicy, std::allocator<o2::framework::ResourcePolicy> > const&, std::vector<o2::framework::CallbacksPolicy, std::allocator<o2::framework::CallbacksPolicy> > const&, std::vector<o2::framework::SendingPolicy, std::allocator<o2::framework::SendingPolicy> > const&, std::vector<o2::framework::ConfigParamSpec, std::allocator<o2::framework::ConfigParamSpec> > const&, o2::framework::ConfigContext&)
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/tests/o2-test-framework-SimpleDataProcessingDevice01() [0x40e1a5]
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/tests/o2-test-framework-SimpleDataProcessingDevice01() [0x40929f]
[25871:internal-dpl-injected-dummy-sink]:     /lib64/libc.so.6: __libc_start_main()
[25871:internal-dpl-injected-dummy-sink]:     /sw/BUILD/3cf74f0b467b5aafdb2b524cc3423ded4e650446/O2/stage/tests/o2-test-framework-SimpleDataProcessingDevice01() [0x40936e]
[25871:internal-dpl-injected-dummy-sink]: Backtrace complete.
```

Thanks a lot!

**Comment by adriansev** [comment]
ping @ktf @TimoWilken .. guys any idea about this error? what would be the path forward? Thanks a lot!!

**Comment by adriansev** [comment]
ping @ktf @TimoWilken Hi guys! please could you take a look at this?

**Pull request merged**